src/plugins/BehaviorSpecGenerator/BehaviorSpecGenerator.js

```diff
@@ -232,18 +234,29 @@ define([
             componentTypeNames = {},
             name,
             nodePath,
+            totalguardNames = {},
```

Comment:
This method is getting too long... I would extract the checking of ComponentTypes in a separate method.